src/sentry/integrations/bitbucket_server/webhook.py

```diff
@@ -49,35 +46,39 @@ def update_repo_data(self, repo, event):
             repo.update(name=name_from_event, config=dict(repo.config, name=name_from_event))
 
 
-class PushEventWebhook(Webhook):
+class PushEventWebhook(BitbucketServerWebhook):
     @property
     def event_type(self) -> IntegrationWebhookEventType:
         return IntegrationWebhookEventType.PUSH
 
-    def __call__(
-        self, organization: Organization, integration_id: int, event: Mapping[str, Any]
-    ) -> HttpResponse:
+    def __call__(self, event: Mapping[str, Any], **kwargs) -> None:
         authors = {}
 
+        if not (
+            (organization := kwargs.get("organization"))
+            and (integration_id := kwargs.get("integration_id"))
+        ):
+            raise ValueError("Organization and integration_id must be provided")
+
         try:
             repo = Repository.objects.get(
                 organization_id=organization.id,
                 provider=PROVIDER_NAME,
                 external_id=str(event["repository"]["id"]),
             )
         except Repository.DoesNotExist:
-            return HttpResponse(status=404)
+            raise Http404()
 
         provider = repo.get_provider()
         try:
             installation = provider.get_installation(integration_id, organization.id)
         except Integration.DoesNotExist:
-            return HttpResponse(status=404)
+            raise Http404()
 
         try:
             client = installation.get_client()
         except IntegrationError:
-            return HttpResponse(status=400)
+            raise SuspiciousOperation()
```

Comment:
idk if this is the right error to raise but this will return 400

Reply:
i could also `raise BadRequest()`

Reply:
ApiHostError by default is a 503

Reply:
Interesting, I've never seen `SuspiciousOperation` exceptions before. Is this something we consider potentially malicious, or just an internal consistency issue? (context: https://docs.djangoproject.com/en/5.1/ref/exceptions/#django.core.exceptions.SuspiciousOperation)

If not, raising `BadRequest` seems more appropriate.

Reply:
@GabeVillalobos there's only a few of exceptions that Django automatically parses into 400s, i found `BadRequest` this morning